### ISSUE TITLE
502 Bad Gateway using haos rpi3 64bit

### ISSUE THREAD

**Issue by lightenup** [hassio-addons/addon-grocy#224]
# Problem/Motivation

Ref #197. This issue is now closed, so I'll report my findings here. I see others in the closed issue talked about the same 64 vs 32 bit problem.

When I install a 64 bit HAOS 6.2 for raspberry pi 3 and add the grocy addon version 0.15.0, I get the same problem as reported in #197. When I use the 32 bit version of HOAS everything seems to work fine.

## Expected behavior

>Grocy to open and land on default start screen

## Actual behavior

> Getting error message: 502 Bad Gateway -- nginx

## Steps to reproduce

> Install a 64 bit HAOS 6.2 for raspberry pi 3 and add the grocy addon version 0.15.0. Start addon and go to web UI.

**Comment by Ninuss** [comment]
Same problem here from last core update. My log at grocy addon supervisor page:

```[s6-init] making user provided files available at /var/run/s6/etc...exited 0.
[s6-init] ensuring user provided files have correct perms...exited 0.
[fix-attrs.d] applying ownership & permissions fixes...
[fix-attrs.d] 01-grocy: applying... 
[fix-attrs.d] 01-grocy: exited 0.
[fix-attrs.d] done.
[cont-init.d] executing container initialization scripts...
[cont-init.d] 00-banner.sh: executing... 
-----------------------------------------------------------
 Add-on: Grocy
 ERP beyond your fridge! A groceries & household management solution for your home
-----------------------------------------------------------
 Add-on version: 0.15.0
 You are running the latest version of this add-on.
 System: Home Assistant OS 6.3  (aarch64 / raspberrypi3-64)
 Home Assistant Core: 2021.9.5
 Home Assistant Supervisor: 2021.09.0
-----------------------------------------------------------
 Please, share the above information when looking for help
 or support in, e.g., GitHub, forums or the Discord chat.
-----------------------------------------------------------
[cont-init.d] 00-banner.sh: exited 0.
[cont-init.d] 01-log-level.sh: executing... 
[cont-init.d] 01-log-level.sh: exited 0.
[cont-init.d] grocy.sh: executing... 
[10:18:38] INFO: Patching Grocy to fix relative URL handling...
patching file views/layout/default.blade.php
Hunk #1 succeeded at 88 with fuzz 2 (offset 1 line).
[cont-init.d] grocy.sh: exited 0.
[cont-init.d] nginx.sh: executing... 
[cont-init.d] nginx.sh: exited 0.
[cont-init.d] php-fpm.sh: executing... 
[cont-init.d] php-fpm.sh: exited 0.
[cont-init.d] done.
[services.d] starting services
[services.d] done.
[10:18:44] INFO: Starting PHP-FPM...
[10:19:11] INFO: Starting NGinx....
2021/09/11 01:07:59 [error] 1188#1188: *1 recv() failed (104: Connection reset by peer) while reading response header from upstream, client: 172.30.32.2, server: a0d7b954-grocy, request: "GET / HTTP/1.1", upstream: "fastcgi://127.0.0.1:9002", host: "secret!.duckdns.org:8123", referrer: "https://secret!.duckdns.org:8123/a0d7b954_grocy"
```

**Comment by amorabito** [comment]
Also experiencing this issue using RPi 3 B+, just discovered the add-on and excited to use! Willing to help investigate just not familiar with home assistant development

**Comment by TerrorBite** [comment]
I am also experiencing this issue.

@frenck said in #197,

> Resolved on dev with the recent upgrades for the upcoming Grocy 3.10.

Can we then assume that the next release will fix this issue? When is this release expected?

**Comment by TerrorBite** [comment]
No updated version is yet available, issue still exists.

**Comment by crisjff** [comment]
Same here with 64 bit HAOS 6.6 for raspberry pi 3 - Grocy 0.15.0 - nginx and duck dns installed.

**Comment by adrigrillo** [comment]
Yes, I can also reproduce the problem with a brand new installation of HAOS. My current configuration is as follows:

```
Add-on version: 0.15.0
You are running the latest version of this add-on.
System: Home Assistant OS 6.6  (aarch64 / raspberrypi3-64)
Home Assistant Core: 2021.11.1
Home Assistant Supervisor: 2021.10.8
```

and the log after getting the 502 error:

```
2021/11/09 00:05:09 [error] 739#739: *1 recv() failed (104: Connection reset by peer) while reading response header from upstream, client: 172.30.32.2, server: a0d7b954-grocy, request: "GET / HTTP/1.1", upstream: "fastcgi://127.0.0.1:9002", host: "192.168.0.244:8123", referrer: "http://192.168.0.244:8123/a0d7b954_grocy"
```

**Comment by KalebBolliger** [comment]
Same problem with the same log as the first reply here. 64-bit HAOS on a raspberry pi 3. 
I'm currently on version 3.10, so it seems the fix did not make it to that version. 
I'm wondering if I should just downgrade to 32-bit at this point.

**Comment by kovalcikr** [comment]
I have the same issue (same config RPI3B+ with 64-bit HAOS).
I did some debugging and the the `php-fpm` produces segmentation fault on any request.

Error LOG from `/var/log/php8/error.log` (with debug logging enabled):
```
[13-Nov-2021 20:28:56.887314] DEBUG: pid 525, fpm_scoreboard_init_main(), line 38: got clock tick '100'
[13-Nov-2021 20:28:56.887815] DEBUG: pid 525, fpm_signals_init_main(), line 219: Unblocking all signals
[13-Nov-2021 20:28:56.888031] DEBUG: pid 525, fpm_socket_af_inet_socket_by_addr(), line 298: Found address for 127.0.0.1, socket opened on 127.0.0.1
[13-Nov-2021 20:28:56.888190] DEBUG: pid 525, fpm_socket_af_inet_socket_by_addr(), line 298: Found address for 127.0.0.1, socket opened on 127.0.0.1
[13-Nov-2021 20:28:56.888260] DEBUG: pid 525, fpm_event_init_main(), line 348: event module is epoll and 1 fds have been reserved
[13-Nov-2021 20:28:56.888311] NOTICE: pid 525, fpm_init(), line 83: fpm is running, pid 525
[13-Nov-2021 20:28:56.888350] DEBUG: pid 525, fpm_children_make(), line 407: blocking signals before child birth
[13-Nov-2021 20:28:56.891527] DEBUG: pid 525, fpm_children_make(), line 431: unblocking signals, child born
[13-Nov-2021 20:28:56.891721] DEBUG: pid 525, fpm_children_make(), line 437: [pool ingress] child 686 started
[13-Nov-2021 20:28:56.891830] DEBUG: pid 525, fpm_children_make(), line 407: blocking signals before child birth
[13-Nov-2021 20:28:56.895345] DEBUG: pid 525, fpm_children_make(), line 431: unblocking signals, child born
[13-Nov-2021 20:28:56.895590] DEBUG: pid 525, fpm_children_make(), line 437: [pool ingress] child 687 started
[13-Nov-2021 20:28:56.895655] DEBUG: pid 525, fpm_children_make(), line 407: blocking signals before child birth
[13-Nov-2021 20:28:56.899018] DEBUG: pid 525, fpm_children_make(), line 431: unblocking signals, child born
[13-Nov-2021 20:28:56.899252] DEBUG: pid 525, fpm_children_make(), line 437: [pool ingress] child 688 started
[13-Nov-2021 20:28:56.899344] DEBUG: pid 525, fpm_children_make(), line 407: blocking signals before child birth
[13-Nov-2021 20:28:56.902839] DEBUG: pid 525, fpm_children_make(), line 431: unblocking signals, child born
[13-Nov-2021 20:28:56.903112] DEBUG: pid 525, fpm_children_make(), line 437: [pool www] child 689 started
[13-Nov-2021 20:28:56.903178] DEBUG: pid 525, fpm_children_make(), line 407: blocking signals before child birth
[13-Nov-2021 20:28:56.906856] DEBUG: pid 525, fpm_children_make(), line 431: unblocking signals, child born
[13-Nov-2021 20:28:56.907149] DEBUG: pid 525, fpm_children_make(), line 437: [pool www] child 690 started
[13-Nov-2021 20:28:56.907218] DEBUG: pid 525, fpm_children_make(), line 407: blocking signals before child birth
[13-Nov-2021 20:28:56.910758] DEBUG: pid 525, fpm_children_make(), line 431: unblocking signals, child born
[13-Nov-2021 20:28:56.911029] DEBUG: pid 525, fpm_children_make(), line 437: [pool www] child 691 started
[13-Nov-2021 20:28:56.911165] DEBUG: pid 525, fpm_event_loop(), line 377: 23616 bytes have been reserved in SHM
[13-Nov-2021 20:28:56.911204] NOTICE: pid 525, fpm_event_loop(), line 378: ready to handle connections
[13-Nov-2021 20:28:57.912399] DEBUG: pid 525, fpm_pctl_perform_idle_server_maintenance(), line 382: [pool ingress] currently 0 active children, 3 spare children, 3 runni
ng children. Spawning rate 1
[13-Nov-2021 20:28:57.912667] DEBUG: pid 525, fpm_pctl_perform_idle_server_maintenance(), line 382: [pool www] currently 0 active children, 3 spare children, 3 running c
hildren. Spawning rate 1
[13-Nov-2021 20:28:58.914014] DEBUG: pid 525, fpm_pctl_perform_idle_server_maintenance(), line 382: [pool ingress] currently 0 active children, 3 spare children, 3 runni
ng children. Spawning rate 1
...
...
...
[13-Nov-2021 20:29:43.343365] DEBUG: pid 525, fpm_got_signal(), line 82: received SIGCHLD
[13-Nov-2021 20:29:43.343630] DEBUG: pid 525, fpm_event_loop(), line 430: event module triggered 1 events
[13-Nov-2021 20:29:43.343828] WARNING: pid 525, fpm_children_bury(), line 259: [pool www] child 689 exited on signal 11 (SIGSEGV - core dumped) after 46.440747 seconds f
rom start
[13-Nov-2021 20:29:43.343920] DEBUG: pid 525, fpm_children_make(), line 407: blocking signals before child birth
[13-Nov-2021 20:29:43.351684] DEBUG: pid 525, fpm_children_make(), line 431: unblocking signals, child born
[13-Nov-2021 20:29:43.352141] NOTICE: pid 525, fpm_children_make(), line 437: [pool www] child 694 started
[13-Nov-2021 20:29:43.984169] DEBUG: pid 525, fpm_pctl_perform_idle_server_maintenance(), line 382: [pool ingress] currently 0 active children, 3 spare children, 3 runni
ng children. Spawning rate 1
[13-Nov-2021 20:29:43.984399] DEBUG: pid 525, fpm_pctl_perform_idle_server_maintenance(), line 382: [pool www] currently 0 active children, 3 spare children, 3 running c
hildren. Spawning rate 1
```
Unfortunately I cannot find core dump anywhere. The root cause of issue is still missing but maybe this will help someone to invesigate it deeper.

**Comment by kovalcikr** [comment]
> I am also experiencing this issue, and I am also using a 64-bit Pi 3B+.
> 
> @frenck said in #197,
> 
> > Resolved on dev with the recent upgrades for the upcoming Grocy 3.10.
> 
> Can we then assume that the next release will fix this issue? When is this release expected?

Tested on latest dev (commit `ad5c93a0e26c286df9772ca8c57aa35dfb24a73c`) the issue is still there.

**Comment by vivobg** [comment]
Same on Odroid C2 aarch64, so not limited to RPi 3.
 Add-on version: 0.15.0
 System: Home Assistant OS 6.6  (aarch64 / odroid-c2)
 Home Assistant Core: 2021.11.4
 Home Assistant Supervisor: 2021.10.8

**Comment by bocklucas** [comment]
Is there any update on this? Short of moving to a Pi 4?

**Comment by kovalcikr** [comment]
just tested version 0.16.0 an it works! thank you @frenck 
Home Assistant OS 7.0, supervisor-2021.12.2, core-2021.12.3

**Comment by frenck** [comment]
Thanks for confirming @kovalcikr ❤️

**Comment by bocklucas** [comment]
> This should be fixed by <https://github.com/hassio-addons/addon-grocy/pull/257>

Thanks @frenck !!😁

**Comment by crisjff** [comment]
> Same here with 64 bit HAOS 6.6 for raspberry pi 3 - Grocy 0.15.0 - nginx and duck dns installed.

Fixed for me too on version 0.16.0, Home Assistant OS 7.0, core-2021.12.4, supervisor-2021.12.2, Hassio Raspberry Pi 3b+. Thank you.
